test/shoppingcart_test.rb

```diff
@@ -138,8 +132,9 @@ def test_sorted_products_by_quantity
     cart.add_product(product3)
     cart.add_product(product4)
 
-    assert_equal [{:meat => [product2], :paper => [product1, product3], :produce =>[product4]}], cart.product_breakdown
-  end
+    breakdown = {:meat => [product2], :paper => [product1, product3], :produce => [product4]}
 
+    assert_equal breakdown, cart.product_breakdown
```

Comment:
Testing a hash is tricky since the order isn't fixed. I assume that's why you put the hash in an array even though that wasn't what was in the pattern provided? Anyways, by saving it to a variable I was able to run the test no problem.